test/ads.jl

```diff
@@ -0,0 +1,129 @@
+using TaylorIntegration
+using Test
+
+@testset "Automatic Domain Splitting" begin
+    using TaylorIntegration: ADSBinaryNode, countnodes, timesvector
+    using StaticArrays: SVector
+
+    # This example is based upon section 3 of
+    # https://doi.org/10.1007/s10569-015-9618-3.
+
+    # Dynamical function
+    @taylorize function kepler_eqs!(dq, q, params, t)
+        dq[1] = q[3]
```

Comment:
Another interesting check/question can be the following one: write down an initial condition `q1` as above in terms of Cartesian position and velocity, while defining the JT variables in terms of perturbations in the Keplerian elements `δa`, `δe`, etc. Then, upon propagating this initial condition on a domain `dom`, can we expect almost no splitting, since in this setup the elements are preserved?